cfme/tests/infrastructure/test_host.py

```diff
@@ -426,3 +426,29 @@ def test_infrastructure_hosts_navigation_after_download(
     hosts_view.navigation.select("Compute")
     provider_view = provider.create_view(InfraProviderDetailsView)
     assert provider_view.is_displayed
+
+
+@test_requirements.infra_hosts
+@pytest.mark.meta(blockers=[BZ(1746214, forced_streams=["5.10"])], automates=[1746214])
+def test_compare_hosts_from_provider_allhosts(appliance, setup_provider_min_hosts, provider):
+    """
+    Polarion:
+        assignee: prichard
+        casecomponent: Infra
+        caseimportance: high
+        initialEstimate: 1/6h
+    Bugzilla:
+        1746214
+
+    """
+    hosts_view = navigate_to(provider.collections.hosts, "All")
+    ent_slice = slice(0, 2, None)
+    num_hosts = hosts_view.entities.paginator.items_amount
+    if num_hosts < 2:
+        pytest.skip('not enough hosts in appliance UI to run test')
+    for h in hosts_view.entities.get_all(slice=ent_slice):
+        h.check()
+    hosts_view.toolbar.configuration.item_select('Compare Selected items',
+                                                 handle_alert=True)
+    compare_hosts_view = provider.create_view(HostsCompareView)
+    assert compare_hosts_view.is_displayed
```

Comment:
For this test do you also want to make sure that the proper hosts (i.e. the ones you selected) are the hosts you're comparing? I'll leave it as _Optional_

Reply:
While 9456 is under review, consider adding some more assertions here against the HostsCompareView, and more widgets defined in the view if needed.